File: typeface-views/src/main/java/com/sorcererxw/typefaceviews/TypefaceViews.java
package com.sorcererxw.typefaceviews;

import android.graphics.Typeface;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/11/23
 */

public class TypefaceViews {
    public enum TypefaceType {
        MEDIUM,
        REGULAR
    }

    private static Typeface mMediumTypeface = null;
    private static Typeface mRegularTypeface = null;

    public static void install(Typeface mediumTypeface,
                               Typeface regularTypeface) {
        mMediumTypeface = mediumTypeface;
        mRegularTypeface = regularTypeface;
    }

    public static Typeface getMediumTypeface() {
        return mMediumTypeface;
    }

    public static Typeface getRegularTypeface() {
        return mRegularTypeface;
    }
}

File: typeface-views/src/main/java/com/sorcererxw/typefaceviews/TypefaceViews.kt
package com.sorcererxw.typefaceviews

import android.graphics.Typeface

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/11/23
 */

object TypefaceViews {

    var mediumTypeface: Typeface? = null
        private set
    var regularTypeface: Typeface? = null
        private set

    enum class TypefaceType {
        MEDIUM,
        REGULAR
    }

    fun install(mediumTypeface: Typeface?,
                regularTypeface: Typeface?) {
        this.mediumTypeface = mediumTypeface
        this.regularTypeface = regularTypeface
    }
}
